pkg/storage/unified/client.go

```diff
@@ -281,14 +283,22 @@ func grpcConn(address string, metrics *clientMetrics) (*grpc.ClientConn, error)
 	// This reduces the number of requests made to the DNS servers.
 	opts = append(opts, grpc.WithDisableServiceConfig())
 
+	if clientKeepaliveTime > 0 {
+		opts = append(opts, grpc.WithKeepaliveParams(keepalive.ClientParameters{
+			Time:                clientKeepaliveTime,
+			Timeout:             10 * time.Second,
+			PermitWithoutStream: true,
+		}))
+	}
 	// Create a connection to the gRPC server
 	return grpc.NewClient(address, opts...)
 }
 
 // GrpcConn is the public constructor that can be used for testing.
+// TODO: also use grpc_client_keepalive_time here.
```

Comment:
This can be done in another PR and will likely fix any error that gitsync is seeing right now with "goaway".